resources/src/stellar_error.rs

```diff
@@ -0,0 +1,158 @@
+use std::error::Error;
+use std::fmt;
+
+/// A resource for the stellar horizon API specific error codes.
+/// These errors adhere to the Problem Details Standard
+/// https://tools.ietf.org/html/draft-ietf-appsawg-http-problem-00
+/// and a list of possible erros can be found here
+/// https://www.stellar.org/developers/horizon/reference/errors.html
+#[derive(Deserialize)]
+pub struct StellarError {
+    // type is a protected word
+    #[serde(rename = "type")] problem_type: String,
+    title: String,
+    status: u16,
+    detail: String,
```

Comment:
I believe this may be optional
https://www.stellar.org/developers/horizon/reference/errors/not-acceptable.html